MeshLib/Elements/HexRule8.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "HexRule8.h"
+
+#include "logog/include/logog.hpp"
+
+#include "GeoLib/AnalyticalGeometry.h"
+
+#include "MeshLib/Node.h"
+#include "Quad.h"
+#include "Line.h"
+
+namespace MeshLib {
+
+const unsigned HexRule8::n_all_nodes;
+
+const unsigned HexRule8::n_base_nodes;
+
+const unsigned HexRule8::face_nodes[6][4] =
+{
+	{0, 3, 2, 1}, // Face 0
+	{0, 1, 5, 4}, // Face 1
+	{1, 2, 6, 5}, // Face 2
+	{2, 3, 7, 6}, // Face 3
+	{3, 0, 4, 7}, // Face 4
+	{4, 5, 6, 7}  // Face 5
+};
+
+const unsigned HexRule8::edge_nodes[12][2] =
+{
+	{0, 1}, // Edge 0
+	{1, 2}, // Edge 1
+	{2, 3}, // Edge 2
+	{0, 3}, // Edge 3
+	{4, 5}, // Edge 4
+	{5, 6}, // Edge 5
+	{6, 7}, // Edge 6
+	{4, 7}, // Edge 7
+	{0, 4}, // Edge 8
+	{1, 5}, // Edge 9
+	{2, 6}, // Edge 10
+	{3, 7}  // Edge 11
+};
+
+const Element* HexRule8::getFace(const Element* e, unsigned i)
```

Comment:
I wonder if it is possible take a hexahedron here instead of a general element as an argument? In this case the code could be more specific, i.e. the number of face nodes is already known (=4). Or in line 57 one could use ```std::array<Node*, 4>``` for storage instead using ```new```. But I do not know if it is possible.